### PR TITLE
feat(connect): [NMI] Use connector_response_reference_id as reference to merchant

### PR DESCRIPTION
**Pull request by TejasMate**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Make any Payment for connector NMI and see that you are getting "reference_id" field in the payments response of hyperswitch. It should not be null.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
Pls address the CI fails
Run `cargo +nightly` fmt for formatting.

File: crates/router/src/connector/nmi/transformers.rs [hunk 1]
```diff
@@ -279,13 +279,13 @@ impl
             Response::Approved => (
                 Ok(types::PaymentsResponseData::TransactionResponse {
                     resource_id: types::ResponseId::ConnectorTransactionId(
-                        item.response.transactionid,
+                        item.response.transactionid.clone(),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:282]: Add variable `orderid` in struct `StandardResponse` and use it as `connector_response_reference_id`
Refer https://secure.nmi.com/merchants/resources/integration/integration_portal.php#transaction_response_variables @TejasMate

File: crates/router/src/connector/nmi/transformers.rs [hunk 2]
```diff
@@ -279,13 +279,13 @@ impl
...
                     ),
                     redirection_data: None,
                     mandate_reference: None,
                     connector_metadata: None,
                     network_txn_id: None,
-                    connector_response_reference_id: None,
+                    connector_response_reference_id: Some(item.response.transactionid),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:288]:
```suggestion
                    connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

File: crates/router/src/connector/nmi/transformers.rs [hunk 3]
```diff
@@ -372,13 +372,13 @@ impl<T>
...
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:381]:
```suggestion
                    connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

File: crates/router/src/connector/nmi/transformers.rs [hunk 4]
```diff
@@ -425,13 +425,13 @@ impl TryFrom<types::PaymentsResponseRouterData<StandardResponse>>
...
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:434]:
```suggestion
                    connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

File: crates/router/src/connector/nmi/transformers.rs [hunk 5]
```diff
@@ -474,13 +474,13 @@ impl<T>
...
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:483]:
```suggestion
                    connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

File: crates/router/src/connector/nmi/transformers.rs [hunk 6]
```diff
@@ -525,13 +525,13 @@ impl TryFrom<types::PaymentsSyncResponseRouterData<types::Response>>
...
                 resource_id: types::ResponseId::ConnectorTransactionId(
-                    response.transaction.transaction_id,
+                    response.transaction.transaction_id.clone(),
                 ),
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(response.transaction.transaction_id),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:534]:
```suggestion
                connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```

Comment (deepanshu-iiitu) [reply crates/router/src/connector/nmi/transformers.rs:534]: @TejasMate

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/nmi/transformers.rs [hunk 7]
```diff
@@ -525,13 +525,13 @@ impl TryFrom<types::PaymentsSyncResponseRouterData<types::Response>>
...
                 resource_id: types::ResponseId::ConnectorTransactionId(
-                    response.transaction.transaction_id,
+                    response.transaction.transaction_id.clone(),
                 ),
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: item.response.orderid.or(Some(item.response.transactionid)),
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/nmi/transformers.rs:534]:
```suggestion
                connector_response_reference_id: None,
```

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/nmi/transformers.rs (hunk 7 above)
Comment (deepanshu-iiitu) [reply crates/router/src/connector/nmi/transformers.rs:534]: Hey @TejasMate 
Thanks for the quick changes.
Can you also do the below change.

Also in line 348 of this file make it as(StandardResponse Struct) :
`pub orderid: Option<String>,`

Also run the following command before you commit the changes to handle the code formatting
`cargo +nightly fmt`

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
Thanks for your interest in contributing to hyperswitch.
Let us know if you need any assistance from our end.
Also, even if hacktoberfest is over, we should celebrate open source everyday and we are open for more contributions from you.
We would still be rewarding folks with goodies even if the PR gets merged post hacktoberfest.
May the Source be with you!

**Comment by deepanshu-iiitu** [comment]
Hey @TejasMate ,
Are you working on the requested changes?

**Comment by TejasMate** [comment]
@deepanshu-iiitu, I apologize for the delay. I will address all the issues this Sunday once I am free from exams

**Review by ArjunKarthik — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**